includes/utilities.inc

```diff
@@ -183,14 +183,14 @@ function islandora_basic_collection_get_other_parents(AbstractObject $object, Ab
  *   The collection object to add to.
  */
 function islandora_basic_collection_add_to_collection(AbstractObject $new_member, AbstractObject $collection) {
-  // Create hook_islandora_basic_collection_share_migrate()
-  foreach (module_implements('islandora_basic_collection_share_migrate') as $module) {
-    $function = $module . '_islandora_basic_collection_share_migrate';
-    $function($new_member, $collection);
-  }
   $results = $new_member->relationships->get(FEDORA_RELS_EXT_URI, 'isMemberOfCollection', $collection->id);
   if (empty($results)) {
     $new_member->relationships->add(FEDORA_RELS_EXT_URI, 'isMemberOfCollection', $collection->id);
+    // Create hook_islandora_basic_collection_share_migrate().
+    // Since fake:pid appears everywhere for some reason, do not trigger when this is the object.
+    if ($new_member->id != "fake:pid") {
```

Comment:
This is a defined constant, so you probably want to use it to incase it evers gets changed.
https://github.com/Islandora/islandora_solution_pack_collection/blob/7.x/includes/ingest.form.inc#L9

Reply:
Though there is a case to be made for allowing these to invoke the hook and push the filtering to the individual implementations.

Reply:
I can change "fake:pid" to the constant, but I think it's better to filter it here at the source than to force every implementation of the hook to filter it (and probably make the same mistakes each time, even if it is documented).

There is not much value to firing the hook every time a collection is viewed, when its purpose is to fire on an object's changing relationships - I'd say the best choice is to filter here.

Reply:
Re the constant: If I change that line to refer to ISLANDORA_BASIC_COLLECTION_FAKE_PID, I get the following notice when using the hook:

> messageNotice: Use of undefined constant ISLANDORA_BASIC_COLLECTION_FAKE_PID - assumed 'ISLANDORA_BASIC_COLLECTION_FAKE_PID' in islandora_basic_collection_add_to_collection() (line 191 of /var/www/drupal/sites/all/modules/islandora_solution_pack_collection/includes/utilities.inc).

If I put 'ISLANDORA_BASIC_COLLECTION_FAKE_PID' in quotes, then the IF statement doesn't actually work - it fires twice when viewing a collection (because of fake:pid).

If I leave it as 'fake:pid', then it works just fine.

What do you suggest, @whikloj?

Reply:
I wonder if it's perhaps because the constant is defined in the admin form - perhaps it is therefore not available in the utilities file?

Would a module_load_include() be too much in order to access the constant?

Reply:
I have no problem with a `module_load_include()` here or you can move the constant to the `.module` file. 
Developers choice.

Reply:
All right, done with the latest commit!